pkg/clustermanager/provider.go

```diff
@@ -20,29 +20,152 @@ import (
 	"context"
 	"errors"
 
+	"github.com/go-logr/logr"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
 	kindprovider "github.com/kubestellar/kubestellar/clusterprovider/kind"
-	v1alpha1apis "github.com/kubestellar/kubestellar/pkg/apis/logicalcluster/v1alpha1"
+	lcv1alpha1apis "github.com/kubestellar/kubestellar/pkg/apis/logicalcluster/v1alpha1"
 	clusterprovider "github.com/kubestellar/kubestellar/pkg/clustermanager/providerclient"
 )
 
 // Each provider gets its own namespace named prefixNamespace+providerName
 const prefixNamespace = "lcprovider-"
 
-func GetNamespace(providerName string) string {
-	return prefixNamespace + providerName
+type provider struct {
+	name            string
+	providerClient  clusterprovider.ProviderClient
+	c               *controller
+	providerWatcher clusterprovider.Watcher
+	nameSpace       string
```

Comment:
nitpicking: should it be capitalized this way? Isn't the convention `namespace` one word?